cli/tests/integrations/test_node.py

```diff
@@ -103,13 +116,73 @@ def test_node_log_invalid_lines():
                    returncode=1)
 
 
-def _node():
-    returncode, stdout, stderr = exec_command(['dcos', 'node', '--json'])
+def test_node_ssh_master():
+    _node_ssh(['--master'])
 
-    assert returncode == 0
-    assert stderr == b''
 
-    return json.loads(stdout.decode('utf-8'))
+def test_node_ssh_slave():
+    slave_id = mesos.DCOSClient().get_state_summary()['slaves'][0]['id']
+    _node_ssh(['--slave={}'.format(slave_id)])
+
+
+def test_node_ssh_option():
+    stdout, stderr = _node_ssh_output(
+        ['--master', '--option', 'Protocol=0'])
+    assert stdout == b''
+    assert stderr.startswith(b'ignoring bad proto spec')
+
+
+def test_node_ssh_config_file():
+    stdout, stderr = _node_ssh_output(
+        ['--master', '--config-file', 'tests/data/node/ssh_config'])
+    assert stdout == b''
+    assert stderr.startswith(b'ignoring bad proto spec')
+
+
+def test_node_ssh_user():
+    stdout, stderr = _node_ssh_output(
+        ['--master', '--user=bogus', '--option', 'PasswordAuthentication=no'])
+    assert stdout == b''
+    assert stderr.startswith(b'Permission denied')
+
+
+def test_node_ssh_no_agent():
+    pass
+
+
+def _node_ssh_output(args):
+    # ssh must run with stdin attached to a tty
+    master, slave = pty.openpty()
+
+    cmd = ('ssh-agent /bin/bash -c ' +
+           '"ssh-add /host-home/.vagrant.d/insecure_private_key ' +
+           '2> /dev/null && dcos node ssh --option StrictHostKeyChecking=no' +
+           ' {}"').format(' '.join(args))
+    proc = subprocess.Popen(cmd,
+                            stdin=slave,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            preexec_fn=os.setsid,
+                            close_fds=True,
+                            shell=True)
+    os.close(slave)
+
+    # wait for the ssh connection
+    time.sleep(5)
```

Comment:
nvm, i don't really care. if this causes problem with the tests we can change later.